Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM gcr.io/google-appengine/nodejs
+
+# Node 16 is required.
+RUN install_node v16
+
+# Copy application code.
+COPY . /app/
+
+# Install production dependencies.
+RUN yarn install --immutable-cache
+
+# Build the app.
+RUN yarn build
+
+# App port forwarding.
+EXPOSE 8080
+
+# Use yarn to host the app.
+CMD yarn start
```

Comment:
I'm not sure where this image will be used, but `yarn start` serves the app using [webpack-dev-server](https://github.com/webpack/webpack-dev-server), which is recommended to only be used for development.

It also serves the app with React in development mode instead of using the production build created by `yarn build`.